### DISCUSSION TITLE
onSubmit mode doesn't re-validate the fields onChange event

### DISCUSSION THREAD

**Discussion by atulgarg22** [react-hook-form/react-hook-form#5550] [Q&A] [answered]
**Describe the bug**
onSubmit mode doesn't re-validate the fields onChange event

**To Reproduce**
  const { register, formState: { errors }, handleSubmit } = useForm({ mode: 'onSubmit' });

**Codesandbox link (Required)**
https://codesandbox.io/s/material-ui-login-myo01

**Expected behavior**
onChange of input fields, revalidate should happen and error message should go.

- OS: Windows 10
- Browser Chrome
- Version Latest

**Comment by bluebill1049** [comment, marked as answer]
<img width="532" alt="Screen Shot 2021-06-10 at 3 34 48 pm" src="https://user-images.githubusercontent.com/10513364/121470330-65679f80-ca01-11eb-95f8-78bce3749425.png">

you are basically overwrite what's inside react hook form's onChange, that's why it's not working.

**Comment by atulgarg22** [comment]
Thank you for your response Bill. It was very helpful and I could resolve
the issue. Really appreciate it.

Regards
Atul Garg

**Reply by bluebill1049** [reply]
No worries. :)
